public/app/plugins/datasource/opentsdb/migrations.ts

```diff
@@ -0,0 +1,28 @@
+type LegacyAnnotation = {
+  fromAnnotations?: boolean;
+  isGlobal?: boolean;
+  target?: string
+  name?: string;
+};
+
+// this becomes the target in the migrated annotations
+const migrateLegacyAnnotation = (json: LegacyAnnotation) => {
+  // return the target annotation
+  let annotation: LegacyAnnotation = {
+    fromAnnotations: true,
+    target: json.target ?? '',
+    name: json.name ?? '',
+    isGlobal: json.isGlobal ?? false,
+  };
+
+  return annotation;
+};
+
+// eslint-ignore-next-line
+export const prepareAnnotation = (json: any) => {
```

Comment:
Any chance to have a type on this `json` value?

Reply:
This is harder then it looks unfortunately: the best I could manage was `(json: LegacyAnnotation | any)`

Reply:
This is a tough one. The `any` type could be a LegacyAnnotation or an OpenTsdbQuery type and those both have a target attribute that is incompatible with each type, either string or DataQuery type. The examples I was given for this used the `any` type and I have not been able to resolve this. I can set the target type of LegacyAnnotation as string or DataQuery but then that breaks the annotations SDK(see below)
![Screen Shot 2022-08-18 at 5 14 45 PM](https://user-images.githubusercontent.com/25674746/185496561-04034ee1-1cfb-4798-8133-6db0b79b52d7.png)